### PR TITLE
Clan System Added (Test)

### PR DESCRIPTION
**Pull request by CO-RONA**

### CODE REVIEW

File: dashboard/routes/api.js [hunk 1]
```diff
@@ -1560,6 +1574,536 @@ router.post("/whitelist/addGroup", CheckAuth, async function (req, res) {
...
+
+		} 
+		const log = await req.client.addLog({
+			action: "USER_LEFT_CLAN",
+			author: { discord: discordAccount, steam: steamAccount },
+			ip: req.session.user.lastIp,
+			details: { details: moreDetails },
+		});
+		await log.save();
+		return res.json({ status: "ok", message: "Limit Set!" });
```
Comment (11TStudio) [inline dashboard/routes/api.js:1723]: Minor issues but should change the message to something else, I think this is a copy paste issue here :)

File: dashboard/routes/api.js [hunk 2]
```diff
@@ -1560,6 +1574,536 @@ router.post("/whitelist/addGroup", CheckAuth, async function (req, res) {
...
+			req.body.clanID,
+		);
+		const log = await req.client.addLog({
+			action: "CLAN_CREATED",
+			author: { discord: discordAccount, steam: steamAccount },
+			ip: req.session.user.lastIp,
+			details: { details: moreDetails },
+		});
+		await log.save();
+		return res.json({ status: "ok", content: await apps });
```
Comment (11TStudio) [inline dashboard/routes/api.js:1775]: You don't need await here... remove it from the res.json({... content: ~~await~~ apps})

File: dashboard/routes/clans.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+const express = require("express"),
+	utils = require("../utils"),
+	CheckAuth = require("../auth/CheckAuth"),
+	router = express.Router(),
+	version = require("../../package.json").version;
+
+    router.get("/", CheckAuth, async (req, res) => {
+        const allCanSeeRoles = await req.client.getAllCanSee();
```
Comment (11TStudio) [inline dashboard/routes/clans.js:8]: Add a checker here if they can see the page or not...
Something like:
```
const canSee = await req.client.canAccess("clans", req.userInfos.id);
	if (!canSee){
		return res.render("404", {
			userRoles: await req.client.getRoles(req.session.user.id),
			ownerID: req.client.config.owner.id,
			serverID: req.client.config.serverID,
			userDiscord: req.userInfos,
			translate: req.translate,
			currentURL: `${req.client.config.dashboard.baseURL}/${req.originalUrl}`,
		});
	}
```

File: dashboard/routes/clans.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            translate: req.translate,
+            repoVersion: req.client.config.version,
+            currentURL: `${req.client.config.dashboard.baseURL}/${req.originalUrl}`,
+            usage: (process.memoryUsage().heapUsed / 1024 / 1024).toFixed(2),
+            config: req.client.config,
+        })
+    });
+
+    router.get("/:clanName", CheckAuth, async (req, res, next) => {
+        const allCanSeeRoles = await req.client.getAllCanSee();
```
Comment (11TStudio) [inline dashboard/routes/clans.js:28]: As above add checker here...

**Review by 11TStudio — CHANGES_REQUESTED** [review]

File: dashboard/routes/clans.js (hunk 4 above)
Comment (CO-RONA) [reply dashboard/routes/clans.js:28]: actually it has, one line below 
`        if (
            !clan ||
            await req.client.getUsersClan(req.userInfos.steam.steamid) != clanID
        )
`

**Review by 11TStudio — APPROVED** [review]
👍 :)

**Pull request merged**